src/Agent.Plugins/GitCliManager.cs

```diff
@@ -106,17 +111,10 @@ public virtual async Task LoadGitExecutionInfo(AgentTaskPluginExecutionContext c
             ArgUtil.NotNull(gitVersion, nameof(gitVersion));
             context.Debug($"Detect git version: {gitVersion.ToString()}.");
 
-            // Resolve the location of git-lfs.
-            // This should be best effort since checkout lfs objects is an option.
-            // We will check and ensure git-lfs version later
-            gitLfsPath = WhichUtil.Which("git-lfs", require: false, trace: context);
-
-            // Get the Git-LFS version if git-lfs exist in %PATH%.
-            if (!string.IsNullOrEmpty(gitLfsPath))
-            {
-                gitLfsVersion = await GitLfsVersion(context);
-                context.Debug($"Detect git-lfs version: '{gitLfsVersion?.ToString() ?? string.Empty}'.");
-            }
+            // Get the Git-LFS version.
+            gitLfsVersion = await GitLfsVersion(context);
```

Comment:
`gitLfsPath` could still be null if git-lfs isn't installed (on Unix or using system git on Windows) so this still needs to be in the `if (!string.IsNullOrEmpty(gitLfsPath))` and not assert it isn't null, like it was before.

Reply:
Thanks, fixed!